Add Range Operator Convention

huwilkes opened the pull request:
Add a new convention to promote using the half-open range operator over the closed range operator when excluding the final value

huwilkes committed:
Add Range Operator Convention

Add a new convention to promote using the half-open range operator over the closed range operator when excluding the final value

dalemyers commented:
I'm  not sure I understand the value here? In your examples, it's not a style preference. One example is correct and the other is wrong? `x..<y` != `x...y` ?

huwilkes commented:
> I'm not sure I understand the value here? In your examples, it's not a style preference. One example is correct and the other is wrong? `x..<y` != `x...y` ?

Right, but `x..<y` is the same as `x...y-1`. I can remove the one example that's obviously just wrong, but I thought it was worth including. The other examples should all be more or less the same, except the one that's 1...5, since that one prints out the actual numbers instead of strings.

anandrajeswaran commented on RangeOperators.md:
```diff
@@ -0,0 +1,80 @@
+//five
+//Fatal error: Index is out of range
+
+// bad:
```
Can we explain why this is bad? Something like "unnecessary arithmetic" calling back to one of the phrases you introduced in the Rationale?

anandrajeswaran replied:
Same comment elsewhere

anandrajeswaran commented on RangeOperators.md:
````diff
@@ -0,0 +1,80 @@
+//5
+```
+
+### Repeated Actions
````
This one in particular probably needs an explanation since "repeated actions" is not a term used in the rationale?

huwilkes committed:
Add explanations and remove unnecessary examples

Add explanations for why some examples were good or bad, removed the repeated actions section and the 1...5 example.

anandrajeswaran commented on RangeOperators.md:
```diff
@@ -0,0 +1,60 @@
+
+## Rationale
+
+Much like using `<` instead of `<=`, using the half-open range operator allows you to exclude the final value without the need of additional arithmetic. That being said, if you intend to use the final, the closed range operator is still suitable.
```
`if you intend to use the final` should be `if you intend to use the final value`?

anandrajeswaran commented on RangeOperators.md:
```diff
@@ -0,0 +1,60 @@
+//four
+//five
+
+// good: One-Sided Ranges can also be used to show the range is from the start to just before `count`
```
In general I would say examples should only be there if they help back up convention/rationale. It's not clear to me that this is. Maybe I'm missing something but this just looks like another case of the previous good example as it relates to the convention/rationale of this page?

anandrajeswaran commented:
I'm still not 100% convinced of the value of this convention. I wonder if a more realistic example could help because in this case all the examples would realistically be replaced with .forEach or .map or some more natural way of iterating the array?

huwilkes committed:
Remove One-Sided range operator example

huwilkes committed:
Add missing word

Added the missing "value" in "if you intend to use the final value"

huwilkes committed:
Update the examples

Updated the examples to use the index in the print statement
